cluster-autoscaler/provisioningrequest/checkcapacity/provisioningclass.go

```diff
@@ -95,130 +95,123 @@ func (o *checkCapacityProvClass) Provision(
-		prs = provreqclient.FilterOutProvisioningClass(prs, v1.ProvisioningClassCheckCapacity)
-		if len(prs) == 0 {
-			break
-		}
+	// Add accepted condition to ProvisioningRequests.
+	for _, pr := range prs {
+		conditions.AddOrUpdateCondition(pr.PrWrapper, v1.Accepted, metav1.ConditionTrue, conditions.AcceptedReason, conditions.AcceptedMsg, metav1.Now())
+	}
 
-		// Pick 1 ProvisioningRequest.
-		pr := prs[0]
-
-		scaleUpIsSuccessful, err := o.checkcapacity(prPods, pr)
-		if err != nil {
-			scaleUpStatus, _ := status.UpdateScaleUpError(&status.ScaleUpStatus{}, errors.NewAutoscalerError(errors.InternalError, "error during ScaleUp: %s", err.Error()))
-			combinedStatus.Add(scaleUpStatus)
-		} else if scaleUpIsSuccessful {
-			combinedStatus.Add(&status.ScaleUpStatus{Result: status.ScaleUpSuccessful})
-		} else {
-			combinedStatus.Add(&status.ScaleUpStatus{Result: status.ScaleUpNoOptionsAvailable})
-		}
+	// Check Capacity. Add Provisioned or Failed conditions.
+	processedPrs := o.checkCapacityBatch(prs, &combinedStatus, startTime)
 
-		provisioningRequestsProcessed[pr.Name] = true
-		if stopBatch := o.shouldStopBatchProcessing(len(provisioningRequestsProcessed), startTime); stopBatch {
-			break
-		}
+	// Use client to update ProvisioningRequests conditions.
+	updateRequests(o.client, processedPrs, &combinedStatus)
+
+	return combinedStatus.Export()
+}
 
-		// For batch processing, the next ProvisioningRequest's pods are fetched using the provisioningRequest pods injector.
-		// TODO: Refactor such that injector injects pods for multiple ProvisioningRequests at once. Pods should be segregated by ProvisioningRequest before being processed.
-		prPods, err = o.getNextPrPods(provisioningRequestsProcessed)
-		// Error might be returned here if the pod-template for current ProvisioningRequest is not found in cluster. We continue with the next ProvisioningRequest as it might have a valid pod-template and may be processed successfully.
-		if err != nil {
-			st, _ := status.UpdateScaleUpError(&status.ScaleUpStatus{}, errors.NewAutoscalerError(errors.InternalError, "error during ScaleUp: %s", err.Error()))
-			combinedStatus.Add(st)
+func (o *checkCapacityProvClass) getProvisioningRequestsAndPods(unschedulablePods []*apiv1.Pod) ([]provreq.ProvisioningRequestWithPods, error) {
+	if !o.isBatchEnabled() {
+		klog.Info("Processing single provisioning request (non-batch)")
+		prs := provreqclient.ProvisioningRequestsForPods(o.client, unschedulablePods)
+		prs = provreqclient.FilterOutProvisioningClass(prs, v1.ProvisioningClassCheckCapacity)
+		if len(prs) == 0 {
+			return nil, nil
 		}
+		return []provreq.ProvisioningRequestWithPods{{PrWrapper: prs[0], Pods: unschedulablePods}}, nil
 	}
 
-	return combinedStatus.Export()
+	batch, err := o.provreqInjector.GetCheckCapacityBatch(o.checkCapacityProvisioningRequestMaxBatchSize)
+	klog.Infof("Processing provisioning requests as batch of size %d", len(batch))
+	return batch, err
```

Comment:
Skip the log in case of error:
```suggestion
	batch, err := o.provreqInjector.GetCheckCapacityBatch(o.checkCapacityProvisioningRequestMaxBatchSize)
	if err != nil {
		return nil, err
	}
	klog.Infof("Processing provisioning requests as batch of size %d", len(batch))
	return batch, nil
```